File: 0104-Maximum-Depth-of-Binary-Tree/solution.py
# https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/1033651031/

class Solution:
    def maxDepth(self, node: Optional[TreeNode]) -> int:
        if node == None:
            return 0
        
        l = self.maxDepth(node.left)
        r = self.maxDepth(node.right)

        print(node.val, "returning:", max(l, r) + 1)
        return max(l, r) + 1

File: 0104-Maximum-Depth-of-Binary-Tree/solution.js
# https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/1033652879/

var maxDepth = function(root) {
    if(root == null){
        return 0;
    }

    const l = maxDepth(root.left)
    const r = maxDepth(root.right)

    return Math.max(l, r) + 1
};
